app/lib/actions/katello/content_view_version/export.rb

```diff
@@ -2,35 +2,45 @@ module Actions
-          group_id = "#{org_label}-#{content_view.label}-"\
-                     "v#{content_view_version.major}.#{content_view_version.minor}"
-          action_subject(content_view_version)
-
-          repos = content_view_version.archived_repos.select { |r| r.content_type == 'yum' }
-
-          history = ::Katello::ContentViewHistory.create!(:content_view_version => content_view_version,
-                                                          :user => ::User.current.login,
-                                                          :status => ::Katello::ContentViewHistory::IN_PROGRESS,
-                                                          :action => ::Katello::ContentViewHistory.actions[:export],
-                                                          :task => self.task)
-
-          plan_action(Katello::Repository::Export, repos, export_to_iso, since, iso_size,
-                                                   group_id)
-          plan_self(:history_id => history.id)
+        input_format do
+          param :smart_proxy_id, Integer
+          param :exporter_data, Hash
         end
 
-        def humanized_name
-          _("Export")
+        output_format do
+          param :exported_file_name, String
+          param :exported_file_checksum, String
         end
 
-        def finalize
-          history = ::Katello::ContentViewHistory.find(input[:history_id])
-          history.status = ::Katello::ContentViewHistory::SUCCESSFUL
-          history.save!
+        def plan(content_view_version)
+          unless File.directory?(Setting['pulp_export_destination'])
+            fail ::Foreman::Exception, N_("Unable to export, 'pulp_export_destination' setting is not set to a valid directory.")
+          end
+
+          sequence do
+            smart_proxy = SmartProxy.pulp_master!
+            action_output = plan_action(::Actions::Pulp3::ContentViewVersion::Export,
+                                   content_view_version_id: content_view_version.id,
+                                   smart_proxy_id: smart_proxy.id,
+                                   export_base_dir: Setting['pulp_export_destination']).output
+
+            plan_self(exporter_data: action_output[:exporter_data], smart_proxy_id: smart_proxy.id)
+
+            plan_action(::Actions::Pulp3::ContentViewVersion::DestroyExporter,
+                          smart_proxy_id: smart_proxy.id,
+                          exporter_data: action_output[:exporter_data])
+          end
+        end
+
+        def run
+          smart_proxy = SmartProxy.find(input[:smart_proxy_id])
+          api = ::Katello::Pulp3::Api::Core.new(smart_proxy)
+          export_data = api.export_api.list(input[:exporter_data][:pulp_href]).results.first
+          output[:exported_file_name], output[:exported_file_checksum] = export_data.output_file_info.first
+        end
+
+        def humanized_name
+          _("Export")
```

Comment:
Haven't tested yet, but if this shows up as 'Export' in the task list then we should give some more details. I've seen in other tasks we show product and repo names where appropriate. Maybe we can show the CV name and version?

Reply:
Shouldnt that information be in the input params anyway ?